### ISSUE TITLE
Error relocating /usr/bin/mono: getrandom: symbol not found

### ISSUE THREAD

**Issue by scottdmoore** [frol/docker-alpine-mono#9]
When trying to build this image, recently, I have encountered this error:

```Error relocating /usr/bin/mono: getrandom: symbol not found
The command '/bin/sh -c apk add --no-cache mono --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing &&    apk add --no-cache --virtual=.build-dependencies ca-certificates &&    cert-sync /etc/ssl/certs/ca-certificates.crt &&    apk del .build-dependencies' returned a non-zero code: 127
```

Dockerfile is:
```FROM alpine:3.8
  RUN apk add --no-cache mono --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing && \
    apk add --no-cache --virtual=.build-dependencies ca-certificates && \
    cert-sync /etc/ssl/certs/ca-certificates.crt && \
    apk del .build-dependencies
```

Build output is:
```
Sending build context to Docker daemon  192.4MB
Step 1/2 : FROM alpine:3.8
 ---> 196d12cf6ab1
Step 2/2 : RUN apk add --no-cache mono --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing &&    apk add --no-cache --virtual=.build-dependencies ca-certificates &&    cert-sync /etc/ssl/certs/ca-certificates.crt &&    apk del .build-dependencies
 ---> Running in 7e8121548456
fetch http://dl-cdn.alpinelinux.org/alpine/edge/testing/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
(1/12) Installing libbz2 (1.0.6-r6)
(2/12) Installing expat (2.2.5-r0)
(3/12) Installing libffi (3.2.1-r4)
(4/12) Installing gdbm (1.13-r1)
(5/12) Installing ncurses-terminfo-base (6.1_p20180818-r1)
(6/12) Installing ncurses-terminfo (6.1_p20180818-r1)
(7/12) Installing ncurses-libs (6.1_p20180818-r1)
(8/12) Installing readline (7.0.003-r0)
(9/12) Installing sqlite-libs (3.24.0-r0)
(10/12) Installing python2 (2.7.15-r1)
(11/12) Installing libgcc (6.4.0-r9)
(12/12) Installing mono (5.10.0.179-r1)
Executing busybox-1.28.4-r1.trigger
OK: 271 MiB in 25 packages
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
(1/2) Installing ca-certificates (20171114-r3)
(2/2) Installing .build-dependencies (0)
Executing busybox-1.28.4-r1.trigger
Executing ca-certificates-20171114-r3.trigger
OK: 271 MiB in 27 packages
Error relocating /usr/bin/mono: getrandom: symbol not found
The command '/bin/sh -c apk add --no-cache mono --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing &&    apk add --no-cache --virtual=.build-dependencies ca-certificates &&    cert-sync /etc/ssl/certs/ca-certificates.crt &&    apk del .build-dependencies' returned a non-zero code: 127
```

This is reproducible on multiple systems one with docker version 17 and one with 18.

**Comment by frol** [comment]
The fix is to use musl from the edge as well:

```
apk add --no-cache musl\>1.1.20 --repository http://dl-cdn.alpinelinux.org/alpine/edge/main
```

**Comment by frol** [comment]
I am not going to switch my image to `alpine:edge` or update `musl` from the `edge` repo since it may introduce some unexpected behavior. `musl` will eventually get updated (most probably with Alpine 3.9 release), so the issue will get resolved by itself.

**Comment by scottdmoore** [comment]
Thank you for the fix. I appreciate your time.

**Comment by vhlongmore** [comment]
> The fix is to use musl from the edge as well:
> 
> ```
> apk add --no-cache musl\>1.1.20 --repository http://dl-cdn.alpinelinux.org/alpine/edge/main
> ```

I ran into the same issue; this fix worked for me.

**Comment by frol** [comment]
Resolved with Alpine 3.9 release (https://github.com/frol/docker-alpine-mono/commit/841330213be9511435a747b535c278cadb3a77e6)
